Analysis/Tasks/PWGHF/HFTreeCreatorD0ToKPi.cxx

```diff
@@ -0,0 +1,263 @@
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+/// \file HFTreeCreator2Prong.cxx
+/// \brief Writer of the 2 prong candidates in the form of flat tables to be stored in TTrees.
+///        Intended for debug or for the local optimization of analysis on small samples.
+///        In this file are defined and filled the output tables
+///
+/// \author Nicolo' Jacazio <nicolo.jacazio@cern.ch>, CERN
+
+#include "Framework/runDataProcessing.h"
+#include "Framework/AnalysisTask.h"
+#include "DetectorsVertexing/DCAFitterN.h"
+#include "AnalysisDataModel/HFSecondaryVertex.h"
+#include "AnalysisDataModel/HFCandidateSelectionTables.h"
+#include "AnalysisCore/trackUtilities.h"
+#include "ReconstructionDataFormats/DCA.h"
+
+using namespace o2;
+using namespace o2::framework;
+using namespace o2::aod::hf_cand_prong2;
+
+namespace o2::aod
+{
+namespace full
+{
+DECLARE_SOA_COLUMN(RSecondaryVertex, rSecondaryVertex, float);
+DECLARE_SOA_COLUMN(PtProng0, ptProng0, float);
+DECLARE_SOA_COLUMN(PProng0, pProng0, float);
+DECLARE_SOA_COLUMN(ImpactParameterNormalised0, impactParameterNormalised0, float);
+DECLARE_SOA_COLUMN(PtProng1, ptProng1, float);
+DECLARE_SOA_COLUMN(PProng1, pProng1, float);
+DECLARE_SOA_COLUMN(ImpactParameterNormalised1, impactParameterNormalised1, float);
+DECLARE_SOA_COLUMN(CandidateSelFlag, candidateSelFlag, int8_t);
+DECLARE_SOA_COLUMN(M, m, float);
+DECLARE_SOA_COLUMN(Pt, pt, float);
+DECLARE_SOA_COLUMN(P, p, float);
+DECLARE_SOA_COLUMN(Eta, eta, float);
+DECLARE_SOA_COLUMN(Phi, phi, float);
+DECLARE_SOA_COLUMN(Y, y, float);
+DECLARE_SOA_COLUMN(E, e, float);
+DECLARE_SOA_COLUMN(NSigTPC_Pi_0, nsigTPC_Pi_0, float);
+DECLARE_SOA_COLUMN(NSigTPC_Ka_0, nsigTPC_Ka_0, float);
+DECLARE_SOA_COLUMN(NSigTOF_Pi_0, nsigTOF_Pi_0, float);
+DECLARE_SOA_COLUMN(NSigTOF_Ka_0, nsigTOF_Ka_0, float);
+DECLARE_SOA_COLUMN(NSigTPC_Pi_1, nsigTPC_Pi_1, float);
+DECLARE_SOA_COLUMN(NSigTPC_Ka_1, nsigTPC_Ka_1, float);
+DECLARE_SOA_COLUMN(NSigTOF_Pi_1, nsigTOF_Pi_1, float);
+DECLARE_SOA_COLUMN(NSigTOF_Ka_1, nsigTOF_Ka_1, float);
+DECLARE_SOA_COLUMN(DecayLength, decayLength, float);
+DECLARE_SOA_COLUMN(DecayLengthXY, decayLengthXY, float);
+DECLARE_SOA_COLUMN(DecayLengthNormalised, decayLengthNormalised, float);
+DECLARE_SOA_COLUMN(DecayLengthXYNormalised, decayLengthXYNormalised, float);
+DECLARE_SOA_COLUMN(CPA, cpa, float);
+DECLARE_SOA_COLUMN(CPAXY, cpaXY, float);
+DECLARE_SOA_COLUMN(Ct, ct, float);
+DECLARE_SOA_COLUMN(ImpactParameterProduct, impactParameterProduct, float);
+DECLARE_SOA_COLUMN(CosThetaStar, cosThetaStar, float);
+DECLARE_SOA_COLUMN(MCflag, mcflag, uint8_t);
```

Comment:
This should be `int8_t`. MC flags are negative for antiparticles.

Reply:
Indeed, thanks!